store/auth.go

```diff
@@ -158,3 +160,40 @@ func ReadStoreToken() (*StoreToken, error) {
 
 	return &readStoreToken, nil
 }
+
+// RequestPackageAccessMacaroon requests a macaroon for accessing package data from the ubuntu store
+func RequestPackageAccessMacaroon() (string, error) {
+	emptyJsonData := "{}"
+	req, err := http.NewRequest("POST", myappsPackageAccessAPI, strings.NewReader(emptyJsonData))
+	req.Header.Set("accept", "application/json")
+	req.Header.Set("content-type", "application/json")
+	if err != nil {
+		return "", err
+	}
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	// check return code, error on anything !200
+	if !httpStatusCodeSuccess(resp.StatusCode) {
+		// unexpected result, bail
+		return "", fmt.Errorf("failed to get store macaroon: %v (%v)", resp.StatusCode, resp)
```

Comment:
There isn't any specific error responses in this endpoint, because it is not authenticated and there is no user data sent. In any case, besides pushing updates for the other comments, I'm reworking this a little different.